react-js/react-hooks.md

```diff
@@ -2,89 +2,284 @@
+This will walk through one possible approach to solving this using React hooks. Note that there is no one right answer or one right approach! 
+
+Please fork [this template](https://codepen.io/dpenny/pen/rNVVZrb) and follow along with this tutorial.
+
+0. **What should the component look like?**
+We can start with a 'sketch' of what our component might look like. This does not need UI or styling. We start with an outline so that we can think about what information and functionality this component will need. 
+
+For example,
+``` 
+const {useState, useEffect} = React;
+
+const TodoApp = () => {
+  return (
+    <div>
+      <h3>status here</h3>
+      <div>
+        <input/>
+        <button>
+          Add item
+        </button>
+      </div>
+      
+      <h2> My Todos:</h2>
+        <li>item1</li>
+        <li>item2</li>
+      
+
+      <button> Clear all </button>
+      <button> All done </button>
+    </div>
+  );
+};
+```
+
+2. **What information do we need? (aka what states do we need?)**
+
+We want to store a list of todos and a status message. We need to store the 'new item' value that the user is typing into the 'Add todo' input field
+
+At the top of the component we can use `useState` to do so:
+
+```
+const TodoApp = () => {
+  const [items, setItems] = useState([])
+  const [status, setStatus] = useState('')
+  const [newTodo, setNewTodo] = useState('');
+
+  return (...)
+  }
+```
+
+3. **When should this information update?**
+We now think about when we want these states to update. 
+
+*Adding a Todo*
+- When we type in the input field, we should store that value as 'newTodo`
+- When we press the button add a todo item, the `items` list should update (that `newTodo` item should be appended to `items`)
+-  After we add this item, the input field should clear (`newTodo` should be reset to `''`)
+
+*Clearing Todos*
+- When a user presses 'all done', `items` should be reset (set to `[]`), and `status` should be updated to be `Good Job!`
+- When the user presses 'clear all', `items` should be reset (set to `[]`)
+
+*Too many items*
+When `items.length` is more than 5, the message should update to 'Try to complete current tasks before adding new ones!'
+
+This could look like: 
+```
+const {useState, useEffect} = React;
+
+const TodoApp = () => {
+  const [items, setItems] = useState(['Item one']);
+  const [status, setStatus] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+
+  return (
+    <div>
+      <h3>{status}</h3>
+      <div>
+        <input
+          value={newTodo}
+          onChange={e => {
+	        // We want to set newTodo to be the input field value
+            setNewTodo(e.target.value);
+          }}
+        />
+        <button
+          onClick={() => {
+            // We want newTodo to be appended to items, and we want to setItems to be that new list. 
+            // We also want to reset the input field value
+            setItems([...items, newTodo]);
```

Comment:
Hmm, this is the exact case that `useReducer` solves, I get we don't want to introduce that in the lesson, but I'm not sure modeling this way of using `useState` is the right thing to do either. Thoughts, @alodahl?

Reply:
i wanted to start with a `useState` example because I think seeing different examples with `useState` will help them understand what it does (Even if useReducer is more efficient)- at the end there's a useReducer example, but I want them to feel comfortable with useState and useEffect before useReducer - thoughts?